firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -432,7 +433,9 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
 
     private func selectTabWithSession(tab: Tab, previous: Tab?, sessionData: Data?) {
         assert(Thread.isMainThread, "Currently expected to be called only on main thread.")
-        selectedTab?.createWebview(with: sessionData)
+        let configuration: WKWebViewConfiguration = tab.isPrivate ? self.privateConfiguration : self.configuration
+
+        selectedTab?.createWebview(with: sessionData, configuration: configuration)
```

Comment:
I'm not deeply familiar with all of the potential ramifications of this, it seems like the kind of change that has the potential for hidden side effects (but I might just be hesitant because I'm thinking of all of the blank homepage bugs we've hit lately which were introduced in surprising ways). Might be nice to get extra eyes on this. Approving since this bug seemed urgent; I did a quick test run on the branch and it LGTM so far.

Reply:
I agree, any changes in this area are a little scary. I've proposed in the original tread that we don't release this immediately to give time for issues to surface. The issue in it's current incarnation only impacts users who have enabled the persist private tabs option so I don't think it should be too urgent.